ezcontent_publish.module

```diff
@@ -87,7 +87,8 @@ function ezcontent_publish_form_shortcut_set_customize_form_alter(&$form, $form_
       $form['shortcuts']['links'][$key]['operations']['#links']['delete']['url'] = Url::fromRoute('ezcontent_publish.shortcut.delete_form', array('shortcut' => $key));
       $currentUserRoles = $currentUser->getRoles();
       if (!in_array('administrator', $currentUserRoles)) {
-        $addShortcutUrl = Url::fromRoute('ezcontent_publish.shortcut.link_add', array('shortcut_set' => 'shortcut_set_' . $currentUser->getUsername() . '_' . $currentUser->id()));
+        $currentUrl = ltrim(\Drupal::service('path.current')->getPath(), '/');
+        $addShortcutUrl = Url::fromRoute('ezcontent_publish.shortcut.link_add', array('shortcut_set' => 'shortcut_set_' . $currentUser->getUsername() . '_' . $currentUser->id()), ['query' => ['destination' => $currentUrl]]);
```

Comment:
- Only pass param if user add shortcut from dashboard , otherwise let it behave as default.